backend/src/main/java/com/google/sps/notifs/EmailNotifier.java

```diff
@@ -0,0 +1,182 @@
+    return new AuthorizationCodeInstalledApp(flow, receiver).authorize("user");
+  }
+
+  /**
+   * Create a message from an email.
+   *
+   * @param emailContent Email to be sent to raw of message
+   * @return a message containing a base64url encoded email
+   * @throws IOException if an error occurs writing to the stream
+   * @throws MessagingException for other failures
+   */
+  public static Message createMessageWithEmail(MimeMessage emailContent)
+      throws MessagingException, IOException {
+    ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+    emailContent.writeTo(buffer);
+    byte[] bytes = buffer.toByteArray();
+    String encodedEmail = Base64.encodeBase64URLSafeString(bytes);
+    Message message = new Message();
+    message.setRaw(encodedEmail);
+    return message;
+  }
+
+  /**
+   * Create a MimeMessage using the parameters provided.
+   *
+   * @param toEmail email address of the receiver
+   * @param subject subject of the email
+   * @param bodyText body text of the email
+   * @return the MimeMessage to be used to send email
+   * @throws MessagingException if there was a problem accessing the Store
+   */
+  public MimeMessage createEmail(String toEmail, String subject, String bodyText)
+      throws MessagingException {
+
+    Properties props = new Properties();
+    Session session = Session.getDefaultInstance(props, /* authenticator= */ null);
+
+    MimeMessage email = new MimeMessage(session);
+
+    email.setFrom(new InternetAddress(APPLICATION_EMAIL));
+    email.addRecipient(RecipientType.TO, new InternetAddress(toEmail));
+    email.setSubject(subject);
+    email.setText(bodyText);
+    return email;
+  }
+
+  /** Function that access its api and using it sends an email */
+  //   TODO(#35): Create a dummy email for ad lib itself to send emails.
+  //   TODO(#36): Replace body to send real link to user instead of generic.
+
+  public void notifyUser() throws MessagingException, IOException, GeneralSecurityException {
+    NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+    MimeMessage email =
+        createEmail(
+            getToEmail(),
+            "Ad-Lib Meeting Found",
+            " Hey "
+                + getRecipientName()
+                + " Please Join your Ad-Lib meeting via the link below : \n"
+                + " http://meet.google.com/new");
+    Message messageWithEmail = createMessageWithEmail(email);
+    Gmail service =
+        new Gmail.Builder(HTTP_TRANSPORT, JSON_FACTORY, getCredentials(HTTP_TRANSPORT))
+            .setApplicationName(APPLICATION_NAME)
+            .build();
+    service.users().messages().send("me", messageWithEmail).execute();
+  }
+
+  public String getToEmail() {
+    return toEmail;
+  }
+
+  /** Getter function that returns the string representing the email recipients name . */
+  public String getRecipientName() {
+    return recipientName;
+  }
+
+  public Gmail getService() {
+    return service;
+  }
```

Comment:
please remove these getters unless they're needed by calling production code

Reply:
I am not using them within the email notifier Itself but I am currently using all three of these getters within the tests .

Reply:
Right, since they're not used by any of the non-test code we want to avoid needlessly exposing the internals of our class. Since we have a way to get the toEmail and recipientName from the sent email on the Gmail mocks and the gmail service is instantiated in the test.